sw/device/sca/sha3_serial.c

```diff
@@ -162,53 +162,39 @@ static uint32_t calculate_rate_bits(uint32_t security_level) {
 /**
  * Starts KMAC message without sending START command.
  *
- * Based on dif_kmac_mode_kmac_start().
+ * Based on dif_kmac_mode_sha3_start().
  *
- * Unlike dif_kmac_mode_kmac_start(), this function doesn't provide the START
- * command to the hardware, i.e., just the key is provided and the initial setup
- * for starting a new message is performed.
+ * Unlike dif_kmac_mode_sha3_start(), this function doesn't provide the START
+ * command to the hardware.
  */
-static dif_result_t kmac_msg_start(dif_kmac_mode_kmac_t mode, size_t l,
-                                   const dif_kmac_key_t *k,
-                                   const dif_kmac_customization_string_t *s) {
-  if (k == NULL || l > kDifKmacMaxOutputLenWords) {
-    return kDifBadArg;
-  }
-
-  // Set key strength and calculate rate (r).
+static dif_result_t sha3_msg_start(dif_kmac_mode_sha3_t mode) {
+  // Set key strength and calculate rate (r) and digest length (d) in 32-bit
```

Comment:
We already talked about this, but perhaps s.o has also a comment on this:
set key strength might be misleading, as there is no key in sha3. I guess this is a problem with the dif. Is there a better wording for "kstrength"? Does k stand for key or kmac?